interfaces/seccomp/backend.go

```diff
@@ -169,19 +197,28 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	if err := os.MkdirAll(dir, 0755); err != nil {
 		return fmt.Errorf("cannot create directory for seccomp profiles %q: %s", dir, err)
 	}
-	_, _, err = osutil.EnsureDirState(dir, glob, content)
+	changed, removed, err := osutil.EnsureDirState(dir, glob+".src", content)
 	if err != nil {
 		return fmt.Errorf("cannot synchronize security files for snap %q: %s", snapName, err)
 	}
+	for _, c := range removed {
```

Comment:
This is very nice. We might even split this out in case the branch has any issues with landing since I think this fixes. a real mistake.